contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpScanSpec.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.http;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import org.apache.drill.shaded.guava.com.google.common.base.Joiner;
+import org.apache.drill.shaded.guava.com.google.common.base.MoreObjects;
+
+@JsonTypeName("http-scan-spec")
+public class HttpScanSpec {
+  private final String database;
+
+  private final Map<String, Object> args = new HashMap<>();
+
+  private final String tableName;
+
+  private final String schemaName;
+
+  private final HttpStoragePluginConfig config;
+
+  @JsonCreator
+  public HttpScanSpec(@JsonProperty("schemaName") String schemaName,
+                      @JsonProperty("database") String database,
+                      @JsonProperty("tableName") String tableName,
+                      @JsonProperty("plugin") HttpStoragePlugin plugin) {
```

Comment:
This won't work: the storage plugin is not Jackson serializable. You can use the `@JsonInject` feature to inject it. However, I'm not sure it is even needed here. See other plugins for an example.

Or, since the goal is to include the config, just pass that directly. Though, the config is not really needed: both the group scan and sub scan should already have a copy.

Reply:
I went through and updated this so that it is passing the config instead of the plugin.

Reply:
Fixed this.